src/string-calculator.ts

```diff
@@ -1,112 +1,99 @@
                 this.isUnexpectedEOLAdded = true;
             }
-            this.lastErrorCode = ErrorCode.LastNumberMissedErrorCode;
+            this.lastErrorCode = ErrorCode.LastNumberMissed;
             return this.errorMessage;
         }],
-        [ErrorCode.NegativeNumberErrorCode, (number: string): string => {
-            if (this.lastErrorCode === ErrorCode.NegativeNumberErrorCode) {
+        [ErrorCode.NegativeNumber, (number: string): string => {
+            if (this.lastErrorCode === ErrorCode.NegativeNumber) {
                 this.errorMessage = this.errorMessage.concat(", " + number);
             } else {
                 if (this.errorMessage.trim().length != 0) {
                     this.errorMessage = this.errorMessage.concat('\n');
                 }
                 this.errorMessage = this.errorMessage.concat(`Negative not allowed : ` + number);
             }
-            this.lastErrorCode = ErrorCode.NegativeNumberErrorCode;
+            this.lastErrorCode = ErrorCode.NegativeNumber;
             return this.errorMessage;
         }],
-        [ErrorCode.NumberNotNumericErrorCode, (number: string): string => {
+        [ErrorCode.NumberNotNumeric, (number: string): string => {
             let unexpectedChar = this.getUnexpectedChar(number);
             let position = this.numberList.indexOf(unexpectedChar);
 
             if (this.errorMessage.trim().length != 0) {
                 this.errorMessage = this.errorMessage.concat('\n');
             }
-            this.errorMessage = this.errorMessage.concat(`'${this.customSep}' expected but '${unexpectedChar}' found at position ${position}.`)
+            this.errorMessage = this.errorMessage.concat(`'${this.customSeparator}' expected but '${unexpectedChar}' found at position ${position}.`)
 
-            this.lastErrorCode = ErrorCode.NumberNotNumericErrorCode;
+            this.lastErrorCode = ErrorCode.NumberNotNumeric;
             return this.errorMessage;
         }]
     ]);
 
     public summation(numberList: string): string {
         this.numberList = numberList;
 
-        if (this.numberList.length === 0) {
+        if (!this.numberList) {
             return "0";
         }
 
         let result: number = 0;
-        let regexToSplitListIntoNumbers = this.createRegex();
-        let listOfNumbers: string[] = this.numberList.split(new RegExp(regexToSplitListIntoNumbers));
+        let separatorRegex = this.formSeparatorRegex();
+        let listOfNumbers: string[] = this.numberList.split(new RegExp(separatorRegex));
 
         listOfNumbers.forEach(number => {
-
-            let err: ErrorCode = this.checkForErrors(number);
-            if(!err) {
+            try {
+                this.checkForErrors(number);
                 result += Number(number);
+            } catch (error) {
+                if(error instanceof ErrorCodeExeption) {
+                    this.errorMessage = this.map.get(error.getErrorCode());
```

Comment:
Yes but what will happen if i remove the if part. If other exception is thrown it would be catched here, but we want it to continue up the stack so it will be handled by the catch it is meant to be handled by.

Reply:
What doesn't make sense is that to change the error message in the map handler, then you return it and then you assign it. this doesn't sound good. You either dont return it and change it as a side effect or you return the message and put the update error message logic into the catch clause